pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -59,16 +59,24 @@ func (tc *TidbCluster) PDAutoFailovering() bool {
 	return false
 }
 
-func (tc *TidbCluster) PDRealReplicas() int32 {
+func (tc *TidbCluster) PDStsDesiredReplicas() int32 {
 	return tc.Spec.PD.Replicas + int32(len(tc.Status.PD.FailureMembers))
 }
 
+func (tc *TidbCluster) PDStsActualReplicas() int32 {
+	stsStatus := tc.Status.PD.StatefulSet
+	if stsStatus == nil {
+		return 0
+	}
+	return stsStatus.Replicas
+}
+
 func (tc *TidbCluster) TiKVAllPodsStarted() bool {
-	return tc.TiKVRealReplicas() == tc.Status.TiKV.StatefulSet.Replicas
+	return tc.TiKVDesirdReplicas() == tc.TiKVStsActualReplicas()
 }
 
 func (tc *TidbCluster) TiKVAllStoresReady() bool {
-	if int(tc.TiKVRealReplicas()) != len(tc.Status.TiKV.Stores) {
+	if int(tc.TiKVDesirdReplicas()) != len(tc.Status.TiKV.Stores) {
```

Comment:
Add `Sts`?

Reply:
thanks for catching this, I missed this.

Reply:
fixed